plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/FeatureFlag.kt

```diff
@@ -0,0 +1,69 @@
+package de.fayard.refreshVersions.core
+
+/**
+ * A bug in refreshVersions can break people's builds.
+ *
+ * This is why we are prudent and hide new code behind feature flags
+ * until the feature is deemed to be stable enough for production.
+ *
+ * The user can then opt-in to enable the in-development feature either on the commmand line
+ *
+ * ```bash
+ * ./gradle refreshVersions --enable BUILD_SRC_LIBS
+ * ./gradle refreshVersions --disable GRADLE_UPDATES
+ * ```
+ *
+ * Or permanently in the Settings file
+ *
+ * ```kotlin
+ * refreshVersions {
+ *      featureFlags {
+ *          enable(BUILD_SRC_LIBS)
+ *          disable(GRADLE_UPDATES)
+ *      }
+ * }
+ * ```
+ *
+ * Never delete a flag here since it would break the Settings file
+ * Instead, mark it as @Deprecated("your message here")
+ */
+enum class FeatureFlag(val state: State) {
+    GRADLE_UPDATES(State.ENABLED_BY_DEFAULT),
+    BUILD_SRC_LIBS(State.DISABLED_BY_DEFAULT)
+    ;
+
+    companion object {
+        /**
+         * Where we store the settings coming from the command-line or the Settings file
+         */
+        val userSettings: MutableMap<FeatureFlag, Boolean> = mutableMapOf()
+    }
```

Comment:
Should not be public.
Please, always keep in mind we need to keep the API surface at minimum to avoid leaking details and breaking users.
I think we'll add a way to track API surface evolution at some point so it's easier for you.

Reply:
Good catch
I used @InternalRefreshVersionsApi since it's used in the other plugin